hathor/p2p/netfilter/matches.py

```diff
@@ -0,0 +1,171 @@
+# Copyright 2021 Hathor Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC
+from ipaddress import ip_network
+from typing import TYPE_CHECKING, Any, Set
+
+from structlog import get_logger
+from twisted.internet.defer import Deferred
+from twisted.internet.task import LoopingCall
+
+if TYPE_CHECKING:
+    from hathor.manager import HathorManager
+    from hathor.p2p.netfilter.context import NetfilterContext
+
+logger = get_logger()
+
+
+class NetfilterMatch(ABC):
+    """Abstract match class."""
+    def match(self, context: 'NetfilterContext') -> bool:
+        raise NotImplementedError
+
+
+class NetfilterMatchAll(NetfilterMatch):
+    """Always match."""
+    def match(self, context: 'NetfilterContext') -> bool:
+        return True
+
+
+class NetfilterMatchAnd(NetfilterMatch):
+    """Logic AND operation for two matches. When the first fails, the second is not executed."""
```

Comment:
Didn't understand this part. Doesn't Python Short-Circuit Evaluation enough to prevent second statement from be executed?

Reply:
I'm just explaining how the match behaves in this case. It has exactly the same behavior as the python short-circuit evaluation.